syntax = "proto3";

package recommendation;

service RecommendationService {
  rpc Recommend (RecommendRequest) returns (stream RecommendResponse);
}

message RecommendRequest {
  string user_id = 1;
  string browser_context = 2; // 브라우저 수집 정보(JSON 등)
  string content_type = 3; // 추천 컨텐츠 유형
  string content_period = 4; // 컨텐츠 기간 필터
}

message RecommendResponse {
  string content = 1; // 추천 결과(스트림 chunk)
  string is_final = 2;  // 마지막 chunk 여부
}
